pkg/clusteragent/admission/mutate/agent_sidecar/providers_test.go

```diff
@@ -274,26 +274,20 @@ func TestApplyProviderOverrides(t *testing.T) {
 		t.Run(test.name, func(tt *testing.T) {
 			mockConfig.SetWithoutSource("admission_controller.agent_sidecar.provider", test.provider)
 			mutated, err := applyProviderOverrides(test.basePod)
-			assert.Equal(tt, test.expectMutated, mutated)
 
 			if test.expectError {
 				assert.Error(tt, err)
-			} else {
-				assert.NoError(tt, err)
-
-				if test.expectedPodAfterOverride == nil {
-					assert.Nil(tt, test.basePod)
-				} else {
-					assert.NotNil(tt, test.basePod)
-					assert.Truef(tt,
-						reflect.DeepEqual(*test.basePod, *test.expectedPodAfterOverride),
-						"overrides not applied as expected. expected %v, but found %v",
-						*test.expectedPodAfterOverride,
-						*test.basePod,
-					)
-				}
+				return
 			}
 
+			assert.NoError(tt, err)
```

Comment:
In case of unexpected error at this point, is it worth comparing `mutated` and `test.basePod` in the lines below ?
In case we want to exit the test as soon as we notice the unexpected error, we can use `require`:

```suggestion
			require.NoError(tt, err)
```

Reply:
I changed this one :+1: